src/btf.cpp

```diff
@@ -116,19 +126,10 @@ void BTF::load_kernel_btfs(const std::set<std::string> &modules)
 
     if (mod_name == "vmlinux")
     {
-      btf_objects.push_back(BTFObj{ .btf = btf__load_from_kernel_by_id(id),
-                                    .id = id,
-                                    .name = "vmlinux" });
-      vmlinux_btf = btf_objects.back().btf;
+      btf_objects.front().id = id;
```

Comment:
Maybe add a check that the first (and only) entry in `btf_objects` is for vmlinux?

Reply:
It seems unnecessary, if we reach `BTF::load_kernel_btfs()`, `vmlinux` should be the first entry. Do you mean adding an assertion ?

Reply:
> what if BTF for vmlinux is not the first entry returned by bpf_btf_get_next_id()?

This is OK since we check if the name matched first.

Reply:
You're right, it's not a problem. Thanks for the PR!